projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js

```diff
@@ -4,222 +4,263 @@ import apiFetch from '@wordpress/api-fetch';
+			__( 'Write content of a post titled "%s"', 'jetpack' ),
+			title
+		);
+	};
+
+	const containsAiUntriggeredParapgraph = _content =>
+		_content.filter(
+			block => block.name && block.name === 'jetpack/ai-paragraph' && ! block.attributes.content
+		).length > 0;
+
+	const taxonomies = categoryObjects.filter( cat => cat.id !== 1 ).concat( tagObjects );
 	const categoryNames = taxonomies.map( ( { name } ) => name ).join( ', ' );
+	const contentBefore = useSelect( select => {
+		const editor = select( 'core/block-editor' );
+		const index = editor.getBlockIndex( clientId );
+		return editor.getBlocks().slice( 0, index ) ?? [];
+	} );
 
-	return createPrompt( currentPost.title, editorContent, categoryNames );
-}
+	const content = contentBefore
+		.filter( function ( block ) {
+			return block && block.attributes && block.attributes.content;
+		} )
+		.map( function ( block ) {
+			return block.attributes.content.replaceAll( '<br>', '\n' );
+		} )
+		.join( '\n' );
+	const contentToUseForPrompt = createPrompt( currentPostTitle, content, categoryNames );
+
+	const getSuggestionFromOpenAI = () => {
+		if ( completionFinished || loadingCompletion ) {
+			return;
+		}
 
-export default function Edit( { attributes, setAttributes } ) {
-	const [ loadingCompletion, setLoadingCompletion ] = useState( true );
-	const [ errorMessage, setErrorMessage ] = useState( '' );
-
-	// Here is where we craft the prompt.
-	const formattedPrompt = useSelect( preparePromptBasedOnEditorState, [] );
-
-	//useEffect hook is called only once when block is first rendered.
-	useEffect( () => {
-		//Theoretically useEffect would ensure we only fire this once, but I don't want to fire it when we get data to edit either.
-		if ( ! attributes.content && ! attributes.requestedPrompt ) {
-			getSuggestionFromOpenAI(
-				setAttributes,
-				formattedPrompt,
-				setLoadingCompletion,
-				setErrorMessage
+		setShowRetry( false );
+		setErrorMessage( false );
+		setNeedsMoreCharacters( 0 );
+		setLoadingCompletion( true );
+
+		const data = { content: contentToUseForPrompt };
+		apiFetch( {
+			path: '/wpcom/v2/jetpack-ai/completions',
+			method: 'POST',
+			data: data,
+		} )
+			.then( res => {
+				const result = res.prompts[ 0 ].text;
+
+				// We set it up so it doesn't start with nothing
+				setCompletionFinished( true );
+				setShowAnimation( true );
+				setLoadingCompletion( false );
+				setAttributes( { content: result } );
+			} )
+			.catch( () => {
+				setErrorMessage(
+					__(
+						'Whoops, we have encountered an error. AI is like really, really hard and this is an experimental feature. Please try again later.',
+						'jetpack'
+					)
+				);
+				setLoadingCompletion( false );
+			} );
+	};
+
+	// Waiting state means there is nothing to be done until it resolves
+	const waitingState = completionFinished || loadingCompletion || loadingCategories;
+	const nbCharactersNeeded = numberOfCharactersNeeded - content.length;
+	if ( ! waitingState ) {
```

Comment:
This logic looks wrong to me (or maybe it's just the naming doesn't match the logic).

If it has finished completion, I would think waiting should be false... Shouldn't you only enter the if below when all are true?  So either:

```js
const isCompletelyLoaded = completionFinished && loadingCompletion && loadingCategories;
...
if ( ! isCompletelyLoaded ) { //...
```

or 

```js
// This is the same as ! (completionFinished && loadingCompletion && loadingCategories )
const isWaiting = ! completionFinished || ! loadingCompletion || ! loadingCategories; 
```

Reply:
Hmmm... You might be right here 😒 et me check...

Reply:
Yep. you were right. Here is the fix: 0cac4b13a7ba7c9c7b46278d88a8fddcac85a001